daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/generators/BinaryBooleanCodeGenerator.scala

```diff
@@ -52,13 +53,12 @@ trait BinaryBooleanCodeGenerator extends BinaryValueCodeGenerator {
     val unparseTrueRep = if (e.binaryBooleanTrueRep.isDefined) s"$trueRep" else s"~$falseRep"
 
     val initERDStatement = ""
-    val initSelfStatement = s"    $field = $initialValue;"
     val parseStatement =
-      s"""    parse_$function(&$field, $lengthInBits, $trueRep, $falseRep, pstate);
-         |    if (pstate->error) return;""".stripMargin
+      s"""$indent1$indent2    parse_$function(&$field, $lengthInBits, $trueRep, $falseRep, pstate);
```

Comment:
These templates are getting pretty hard to interpret. really you have to just go look at examples of generated output to see if they make sense. 

That said, I'm not sure how we do better. 

Alternative technique is defining little scala abstract syntax tree nodes  that represent these things, and we construct a structure of those which then generate the nitty gritty code. 
That extra indirection to what is exactly being generated gets in the way of understanding just as much as the cluttered business of these templates now.